### ISSUE TITLE
Ability to use varchar/text sql type to map to rust enum?

### ISSUE THREAD

**Issue by pyros2097** [adwhit/diesel-derive-enum#7]
Is it possible to not have a enum type in the database corresponding to the rust enum as it would be easier to just store the value as a string since it will make migrations easier. That way we wont need to mapping also right? Just serialization and deserialization will suffice.

**Comment by adwhit** [comment]
I think this should work as-is (you would need to keep the mapping the same though). Of course the problem then is that if any text other than the enum is found in the column, it will error when trying to deserialize. Possibly there are ways to make this more lenient, e.g.

```rust
#[derive(PgEnumLenient)]
enum MyText {
    Foo,
    Bar,
    Other<String>    // This catches all other text
}  
```

**Comment by pyros2097** [comment]
Yes it doesn't work. I did find a way to do it myself but forgot. You need to implement a diesel ToSql/FromSql trait and it will actually solve it.

**Comment by adwhit** [comment]
If you figure out how to do this, I will gladly add it to the crate!

**Comment by adwhit** [comment]
Out of scope
